src/language/ast.ts

```diff
@@ -63,15 +82,99 @@ export interface ASTKindToNode {
   ModelDescription: ModelDescriptionNode;
   Name: NameNode;
   OptionalName: OptionalNameNode;
+  // Endpoints types
+  EndpointNamespaceTypeDefinition: EndpointNamespaceTypeDefinitionNode;
+  EndpointVerb: EndpointVerbNode;
+  EndpointUrl: EndpointUrlNode;
+  EndpointTypeDefinition: EndpointTypeDefinitionNode;
+  EndpointResponse: EndpointResponseNode;
+  EndpointSecurityDefinition: EndpointSecurityDefinitionNode;
 }
 
 export interface DocumentNode {
   readonly kind: 'Document';
   readonly loc?: Location;
-  readonly definitions: ReadonlyArray<TypeDefinitionNode>;
+  readonly definitions: ReadonlyArray<TypeDefinitionNode | EndpointNamespaceTypeDefinitionNode>;
```

Comment:
Please, next time don't resolve the thread by yourself if it isn't obvious. We can discuss validation logic on today call. So I ask you to do original suggestion event if validation rule will be broken for now

```
TypeDefinitionNode = EndpointDomainTypeDefinitionNode | ModelDomainTypeDefinitionNode
```